tests/utils/fixtures/fixtures.py

```diff
@@ -0,0 +1,703 @@
+
+    return img_builder
+
+
+@pytest.fixture(scope="session")
+def prepared_test_build_base(request, conversion, bitbake_variables, no_tmp_build_dir):
+
+    if conversion:
+        return {"build_dir": None, "bitbake_corebase": None}
+
+    if no_tmp_build_dir:
+        build_dir = os.environ["BUILDDIR"]
+    else:
+        build_dir = tempfile.mkdtemp(prefix="test-build-", dir=os.environ["BUILDDIR"])
+
+    local_conf = get_local_conf_path(build_dir)
+    local_conf_orig = get_local_conf_orig_path(build_dir)
+    bblayers_conf = get_bblayers_conf_path(build_dir)
+    bblayers_conf_orig = get_bblayers_conf_orig_path(build_dir)
+
+    def cleanup_test_build():
+        if not no_tmp_build_dir:
+            run_verbose("rm -rf %s" % build_dir)
+        else:
+            reset_build_conf(build_dir, full_cleanup=True)
+
+    cleanup_test_build()
+    request.addfinalizer(cleanup_test_build)
+
+    env_setup = "cd %s && . oe-init-build-env %s" % (
+        bitbake_variables["COREBASE"],
+        build_dir,
+    )
+    run_verbose(env_setup)
+
+    if not no_tmp_build_dir:
+        run_verbose("cp %s/conf/* %s/conf" % (os.environ["BUILDDIR"], build_dir))
+        with open(local_conf, "a") as fd:
+            fd.write(
+                'SSTATE_MIRRORS = " file://.* file://%s/PATH"\n'
+                % bitbake_variables["SSTATE_DIR"]
+            )
+            fd.write('DL_DIR = "%s"\n' % bitbake_variables["DL_DIR"])
+
+    run_verbose("cp %s %s" % (local_conf, local_conf_orig))
+    run_verbose("cp %s %s" % (bblayers_conf, bblayers_conf_orig))
+
+    return {"build_dir": build_dir, "bitbake_corebase": bitbake_variables["COREBASE"]}
+
+
+@pytest.fixture(scope="function")
+def prepared_test_build(prepared_test_build_base):
+    """
+    Prepares a separate test build directory where a custom build can be
+    made, which reuses the sstate-cache.
+    """
+
+    reset_build_conf(prepared_test_build_base["build_dir"])
+    return prepared_test_build_base
+
+
+@pytest.fixture(autouse=True)
+def min_mender_version(request, bitbake_variables):
+    version_mark = request.node.get_closest_marker("min_mender_version")
+    if version_mark is None:
+        pytest.fail(
+            (
+                '%s must be marked with @pytest.mark.min_mender_version("<VERSION>") to '
+                + "indicate lowest Mender version for which the test will work."
+            )
+            % str(request.node)
+        )
+
+    test_version = version_mark.args[0]
+    if not version_is_minimum(bitbake_variables, "mender-client", test_version):
+        pytest.skip("Test requires Mender client %s or newer" % test_version)
+
+
+@pytest.fixture(autouse=True)
+def min_yocto_version(request, bitbake_variables):
+    version_mark = request.node.get_closest_marker("min_yocto_version")
+    if version_mark is None:
+        return
+
+    if request.config.getoption("--test-conversion"):
+        pytest.skip("Not applicable for mender-convert")
```

Comment:
I don't think this is correct. Any test in this repo should be run on mender-convert regardless of the `min_yocto_version`. In a way we assume that mender-convert has the same features than latest yocto, with regards to which tests to run.

Keep in mind that yocto specific tests will remain in `meta-mender` and won't be part of the shared module.

Reply:
Hmm, good point. Wondering though :thinking: how were we ever able to get the yocto version for `mender-convert` tests? Or are these tests just simply not meant to be a part of `mender-image-tests`, and should instead be a part of `meta-mender`, like you said? :thinking: :thinking: :thinking:

Reply:
That I can see, there are only two tests in `test_part_image.py` that require min yocto version (warrior, actually, so we could by now remove the marker all together).

I think it just "works" because [this call](https://github.com/mendersoftware/meta-mender/blob/daf755df27d142556e43620c6550b94538757c6a/tests/acceptance/fixtures.py#L507-L519) returns empty when running from mender-convert (or actually from here, as we used to do `cd` into the submodule), therefore the fixture will never skip any test.

Anyway, that is not important now, we only need to remove these two lines that you added and let it be "as-is".

Reply:
It was added due to [this call](https://github.com/mendersoftware/meta-mender/blob/daf755df27d142556e43620c6550b94538757c6a/tests/acceptance/fixtures.py#L525) erroring, due to the index call, when the string is empty. 

So we need to fix this somehow. 

I'm not sure what would be appropriate here though.

Reply:
Interesting, so then this must be related from moving the working dir from inside the submodule (probably checkout with no branch) to the mender-convert dir.

Could we have something like:

```
if yocto_version in yocto_versions_ordered and yocto_versions_ordered.index(test_version) > yocto_versions_ordered.index(
        yocto_version
    ):
```
?

Reply:
Alternatively, just run always:

```suggestion
    if request.config.getoption("--test-conversion"):
        return
```

right? This should achieve the same and it is more meaningful

Reply:
Agreed, second option makes more sense :)